### DISCUSSION TITLE
Lg ThinQ air conditioners.

### DISCUSSION THREAD

**Discussion by delobanov** [nVuln/homebridge-lg-thinq#32] [Ideas]
Hello! Are there any plans to add support for lg air conditioners?

**Comment by nVuln** [comment]
yup, I'm ready to add it when I have device data
please install plugin and turn on homebridge debug, you can see device data in debug logging, upload it here

**Comment by delobanov** [comment]
`Found device:  {"deviceId":"5e3b4465-2db0-1bd3-b6b2-402f8687b008","deviceType":401,"modelName":"RAC_056905_WW","subModelNm":null,"sensorType":null,"alias":"Кондиционер","deviceCode":"AI01","networkType":"02","tftYn":"N","guideTypeYn":"Y","guideType":"RAC_TYPE1","pccModelYn":"N","autoOrderYn":"N","drServiceYn":"N","ssid":"Atmosphere","timezoneCode":"Europe/Moscow","timezoneCodeAlias":"Russia/Moscow","sdsGuide":"{\"deviceCode\":\"AI01\"}","newRegYn":"N","remoteControlType":"","fareTarget":null,"area":"294772","sleep":null,"deviceState":"E","rmsClientId":null,"regDtUtc":"20210316152135000","regIndex":0,"blackboxYn":"Y","groupableYn":"Y","controllableYn":"Y","combinedProductYn":"N","masterYn":"Y","snapshot":{"airState.windStrength":4,"airState.wMode.lowHeating":0,"airState.diagCode":0,"airState.lightingState.displayControl":0,"airState.wDir.hStep":2,"mid":188456900,"airState.energy.onCurrent":74,"airState.wMode.airClean":1,"airState.quality.sensorMon":0,"airState.miscFuncState.antiBugs":0,"airState.tempState.target":22,"airState.operation":1,"airState.wMode.jet":0,"airState.wDir.vStep":0,"timestamp":1628356379251,"airState.powerSave.basic":0,"airState.quality.PM10":21,"static":{"deviceType":"401","countryCode":"RU"},"airState.quality.overall":2,"airState.tempState.current":25,"airState.miscFuncState.extraOp":0,"airState.reservation.sleepTime":0,"airState.miscFuncState.autoDry":0,"airState.reservation.targetTimeToStart":0,"meta":{"allDeviceInfoUpdate":false,"messageId":"6FkrgWUwTPOW_Q7t74PRfg"},"airState.quality.PM1":17,"airState.wMode.smartCare":0,"airState.quality.PM2":19,"online":true,"airState.opMode":2,"airState.reservation.targetTimeToStop":0,"airState.filterMngStates.maxTime":0,"airState.filterMngStates.useTime":0},"manufacture":null,"online":true,"platformType":"thinq2","homeDeviceOrder":-1,"roomDeviceOrder":-1,"ownershipYn":"Y","modelJsonVer":"11.74","modelJsonUri":"https://objectcontent.lgthinq.com/d4976a2c-0d02-4a35-8a4d-1c024b6c18d4?hdnts=exp=1688129626~hmac=e4c8bc0a6bc3f6cb033d016c47b4908715a6e8e85eea77497e78b26cd3271777","appModuleVer":"16.32","appModuleUri":"https://objectcontent.lgthinq.com/6c1a78d3-0404-4712-a5bc-2bc3d9e072b8?hdnts=exp=1685169055~hmac=8b538678344ed708bd929b66403d075c761cc94be01589d82200ab3e93c3fa57","appRestartYn":"Y","appModuleSize":"13305760","langPackProductTypeVer":"51.5","langPackProductTypeUri":"https://objectcontent.lgthinq.com/b2614701-6e1e-47c8-ac83-f3c9b1c8dafc?hdnts=exp=1689819058~hmac=1e94874dfd13c9de71cf38089c781f32b4c98b34a00873864d54001cbfff75b4","langPackModelVer":null,"langPackModelUri":null,"roomId":"161590749430818457","fwInfoList":[{"checksum":"0000479A","partNumber":"SAA40128584","order":1}],"modemInfo":{"modelName":"RAC_056905_WW","appVersion":"clip_hna_v1.9.104","modemType":"QCOM_QCA4010","ruleEngine":"y"},"existsEntryPopup":"N","fwVer":null,"modemVer":null,"subDeviceCount":0,"firebaseLogKey":"P:SD","cardType":"Small","cardControl":"Base"}`

**Comment by delobanov** [comment]
homebridge-lg-thinq-ac produces many errors and doesn't support all device functions. Also it is not work stable

**Reply by nVuln** [reply]
just want to know which other function you want ?
because Homekit is limited, it's hard to implement all device functions, for AC I think on/off ac, manual/auto operation mode, show/adjust temperature, swing, fan speed, and air quality sensor (if available)

**Reply by delobanov** [reply]
You have named all the necessary functions, the main thing is that they work stable.  For example, temperature control in automatic mode.  In homekit, auto mode sets the temperature range, how should the ac process this? Current plug-in just throwing error. Mode switching, fan speed and swing also should work as expected. Ex map fan speed from 0 to 100 with 1-5 ac range. Auto fan speed mode? Should it be in range? Didn’t think so. I don’t know all the  limits of HomeKit ac unit, so please, make it just understandable. Also, please, implement on/off led display on indoor module.

**Comment by nVuln** [comment]
I already published first testing version for AC `v1.1.0-ac.1`, help me test it and following up at #41
